### DISCUSSION TITLE
OPTIONS request method not available?

### DISCUSSION THREAD

**Discussion by dragosMC91** [sindresorhus/got#1766] [Q&A] [answered]
Hello, I am using got to do some data preparation/ cleanup via api for a bunch of e2e tests (webdriverio).

I have been using post, put, delete and get calls so far. Now I have a specific scenario where I need to make an `OPTIONS` request before a POST and I cannot find how.

Am I missing something or is this method not supported ? Can I find a way around it ?

I basically need to extract a specific header from that OPTIONS call and pass it in a POST, otherwise I cannot auth. 

Thanks in advance!

**Comment by sindresorhus** [comment, marked as answer]
There's no convenience method for it (e.g. `got.post`), but you can pass any HTTP method the [`method`](https://github.com/sindresorhus/got/tree/f896aa52abc41fe40d4942da94a0408477358f14#method) option (e.g. `got(url, {method: 'OPTIONS'})`).

**Reply by dragosMC91** [reply]
I found this while re-reading the docs here https://github.com/sindresorhus/got#method.

Thanks for the answer and have a great weekend!
